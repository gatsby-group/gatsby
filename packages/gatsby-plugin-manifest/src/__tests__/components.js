const { generateHeadComponents } = require(`../components`)

describe(`gatsby-plugin-manifest`, () => {
  describe(`generateHeadComponents`, () => {
    it(`creates head components`, () => {
      const headComponents = [];
      for (const component of generateHeadComponents(
        path => path,
        {
          icon: true,
          theme_color: `#000000`,
          manifest: {}
        }
      )) { headComponents.push(component) }
      expect(headComponents).toMatchSnapshot()
    })
    it(`creates prefixed head components`, () => {
      const headComponents = [];
      for (const component of generateHeadComponents(
        path => `/prefix${path}`,
        {
          icon: true,
          theme_color: `#000000`,
          manifest: {}
        }
      )) { headComponents.push(component) }
      expect(headComponents).toMatchSnapshot()
    })
  })
})
