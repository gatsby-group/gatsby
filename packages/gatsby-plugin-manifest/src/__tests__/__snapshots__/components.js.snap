// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest generateHeadComponents creates head components 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest generateHeadComponents creates prefixed head components 1`] = `
Array [
  <link
    href="/prefix/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/prefix/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/prefix/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/prefix/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/prefix/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/prefix/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/prefix/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/prefix/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/prefix/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
  <link
    href="/prefix/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;
